### ISSUE TITLE
Having multiple charts on one page causes accessibility issues

### ISSUE THREAD

**Issue by bourke** [recharts/recharts#3816]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://hs4v2f.csb.app/](https://hs4v2f.csb.app/)

### Steps to reproduce
Load the page and test using an accessibility tester (e.g., https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp/related?utm_source=ext_sidebar&hl=en-GB). 

### What is expected?
The page should pass accessibility tests for WCAG checkpoint 2.4.1 regarding accessible names for ARIA landmarks. 

### What is actually happening?
Because the `recharts-wrapper` `<div>` has a hardcoded `role="region"` without `aria-label`, when multiple charts exist on a page, the ARIA landmarks (`region` is considered a landmark) aren't identified and can't be navigated to quickly using screen readers. A similar phenomenon occurs with tooltip wrappers; they have their roles defined as `dialog` without any identifiers.

| Environment | Info |
|---|---|
| Recharts | v2.8.0 |
| React | 18.2.0 |
| System | macOS 13.6 |
| Browser | Chrome 117.0.5938.132 (arm64) |

---
I realise this issue could be considered a feature request rather than a bug report, but since the code does include ARIA roles without the necessary identifiers, I'm reporting it as a bug. I'll be happy to submit a PR if this isn't on your internal roadmap.

**Comment by nikolasrieble** [comment]
@bourke Thank you for creating this issue. We would love if you could pick this up yourself.

**Comment by julianna-langston** [comment]
It looks like some of the charts have `role="img"`, and some of them have `role="region"`. Would it be easier to replace the sankey plots with `role="img"`, or just to drop the role entirely, rather than trying to juggle aria-label's?

**Comment by ckifer** [comment]
I think in other issues we've discussed that img isn't even the correct role to use on charts anymore right? We should be using the newer spec "graphic-*"?

**Comment by julianna-langston** [comment]
What the "correct" role is open to debate, and many accessibility experts can have reasonable disagreements on it. 

Let me take you through my though process.

When I'm writing code for any reason to go on the web, I add `role` to elements for the sole purpose of communicating something to a screen reader user. Let's go through a couple of options, their pros/cons, and how they would be read by screen readers.

1. `img` - Indicates the element is the equivalent of an <img /> element. Should have an `alt`/equivalent attribute to give a description of the image. Something like `<div role="img" aria-label="Stuff" />` would be read to a screen reader user as "Stuff, graphic". (The exact language and order depends on the screen reader). A user would not expect that they could interact with it. Also, the element would be added to the screen reader's list of graphical items, so users could jump quickly to it if they knew they were looking for an image.
2. `application` - Indicates that the element includes keyboard-based interactions (eg, arrow keys). Depending on the screen reader, the role may not be directly communicated.
3. a landmark role, like `region` - Indicates that the element is the equivalent of a document element like `<section>`. While these can make sense if you have insight into the overall structure of the page, they wind up being noisy and unnecessary if they're used for general content. To quote [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles):
> Too many landmark roles create "noise" in screen readers, making it difficult to understand the overall layout of the page.
4. `graphics-*` - hasn't been picked up by the major screen readers in 6 years, so even the W3C examples just use the `img` role as a fallback.

Ultimately, I'd recommend using `role="img"` for the charts. Sure, it gives screen reader users the impression that the image is static, but in fairness, recharts doesn't support interactions for screen reader users yet. As we make accessibility improvements, I think it's fair to re-evalaute which `role`'s we're using, but for now, `img` seems most appropriate.

**Comment by julianna-langston** [comment]
Note that this still leaves us with a problem around getting `alt`/`aria-label`/equivalent attributes into the system, but that's being addressed in the linked PR.

**Comment by bourke** [comment]
> Note that this still leaves us with a problem around getting `alt`/`aria-label`/equivalent attributes into the system, but that’s being addressed in the linked PR.

It’s worth noting, I think, that changing the `role` to `img` for all charts removes the urgency around the use case that motivated my PR: ensuring that each `region` in a page with multiple `region`s carries a unique accessibility identifier. To your point (3), @julianna-langston, it‘s better to be careful with `region`s in any case.

**Comment by ckifer** [comment]
Thanks for the explanation @julianna-langston I'm terrible at accessibility details. That all makes sense (though I wish screen readers would move to the new attributes). 

Using img makes sense to me, do we want to change the scope of that PR to update everything to img?

**Comment by julianna-langston** [comment]
I'd like to propose a syntax that might address the original concern, with hopefully minimal changes.

For each given categorical chart, we would set a `role="img"` by default, and pass through the properties `role` and `aria-label`. This way, if consumers want to set them for accessibility or testing reasons, they can.

Example using `aria-label`:
```jsx
<LineChart data={data} aria-label="Example">...</LineChart>
```

If a consumer wants to use an experimental role, they could:
```jsx
<LineChart data={data} role="graphics-document img" aria-label="Example">...</LineChart>
```

I would not suggest a default `aria-label` attribute, partially for i18n reasons. This would mean that a rechart component would fail an accessibility scan by default, but that puts it in the same bucket as an img element. If you have a website with `<img src="myGraph.jpg" />`, that will fail until you add alt text: `<img src="myGraph.jpg" alt="AAPL Stock Price" />`. Similarly, in this proposal, you would fail an accessibility scan with `<LineChart data={data}>...</LineChart>`, but you would pass once you added your label: `<LineChart data={data} aria-label="AAPL Stock Price">...</LineChart>`

**Comment by ckifer** [comment]
As long as it's able to be set and is set in the correct place that sounds good to me. We could do all applicable aria attributes if that makes things easier too (unless it makes things worse)
